jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PushBlobStep.java

```diff
@@ -83,8 +85,13 @@ public BlobDescriptor call() throws IOException, RegistryException, ExecutionExc
         return blobDescriptor;
       }
 
-      // todo: leverage cross-repository mounts
-      registryClient.pushBlob(blobDescriptor.getDigest(), blob, null);
+      // If base and target images are in the same registry, then use mount/from to try mounting the
```

Comment:
Haven't looked into the code yet, so asking at a high level: what if the base and target images are in the same registry but are in different repositories requiring different auths? Is that case covered?

Reply:
That's the note below, which I'll reword.

Reply:
> what if the base and target images are in the same registry but are in different repositories requiring different auths?

I don't think this case works.

Below works, showing that I can pull and push to my `francium25` repo.
```xml
          <from>
            <image>francium25/my-test</image>
            <auth>
              <username>francium25</username><password>...</password>
            </auth>
          </from>
          <to>
            <image>francium25/my-test2</image>
            <auth>
              <username>francium25</username><password>...</password>
            </auth>
          </to>
```

Below also works, showing that I can push to another repo of mine, `francium25paran`:
```xml
          <!--from>
            <image>francium25/my-test</image>
            <auth>
              <username>francium25</username><password>...</password>
            </auth>
          </from-->
          <to>
            <image>francium25paran/my-test2</image>
            <auth>
              <username>francium25paran</username><password>...</password>
            </auth>
          </to>
```

However, if I uncomment the `<from>` section above, it doesn't work. The reason is that `francium25` and `francium25paran` require different auths. Checking if they are on the same registry isn't enough.

```
com.google.cloud.tools.jib.registry.RegistryUnauthorizedException: Unauthorized for registry.hub.docker.com/francium25paran/my-test2
        at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:268)
        at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.callWithAllowInsecureRegistryHandling(RegistryEndpointCaller.java:151)
        at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:141)
        at com.google.cloud.tools.jib.registry.RegistryClient.callRegistryEndpoint(RegistryClient.java:367)
        at com.google.cloud.tools.jib.registry.RegistryClient.pushBlob(RegistryClient.java:316)
        at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:94)
        at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:36)
        at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly(CombinedFuture.java:181)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: com.google.api.client.http.HttpResponseException: 401 Unauthorized
{"errors":[{"code":"UNAUTHORIZED","message":"authentication required","detail":[{"Type":"repository","Class":"","Name":"francium25paran/my-test2","Action":"pull"},{"Type":"repository","Class":"","Name":"francium25paran/my-test2","Action":"push"},{"Type":"repository","Class":"","Name":"francium25/my-test","Action":"pull"}]}]}

        at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1070)
        at com.google.cloud.tools.jib.http.Connection.send(Connection.java:161)
        at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:229)
        ... 11 more
```

Reply:
Oh, and for testing, you might want to temporarily force skipping the BLOB availability check.

```
      // check if the BLOB is available
      if (false && registryClient.checkBlob(blobDescriptor.getDigest()) != null) { ...
```

Reply:
I misunderstood.  Major bummer.

Reply:
Yeah, I'm afraid that's a major bummer that may require much more non-trivial changes.